app/renderer/components/navigation/navigator.js

```diff
@@ -169,7 +169,7 @@ class Navigator extends ImmutableComponent {
     if (!e.target.className.includes('navigatorWrapper')) {
       return
     }
-    return !isMaximized() ? currentWindow.maximize() : currentWindow.unmaximize()
+    return !isMaximized() ? windowActions.shouldMaximize(getCurrentWindowId()) : windowActions.shouldMinimize(getCurrentWindowId())
```

Comment:
these are really just temporary to replace the currentWindow.method calls. This should really be `onNavigatorDoubleClick` or something like that